app/screens/plugin/index.tsx

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React from 'react';
+import {shallowEqual, useSelector} from 'react-redux';
+
+import type {GlobalState} from '@mm-redux/types/store';
+
+import {getTheme} from '@mm-redux/selectors/entities/preferences';
+
+import {SafeAreaView} from 'react-native-safe-area-context';
+import {WebView} from 'react-native-webview';
+
+import type {Theme} from '@mm-redux/types/preferences';
+
+import StatusBar from 'app/components/status_bar';
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+
+const HEADERS = {
+    'X-Mobile-App': 'mattermost',
+};
+
+interface PluginProps {
+    link: string;
+}
+
+function Plugin({link}: PluginProps) {
+    const [theme] = useSelector((state: GlobalState) => ([
+        getTheme(state),
+    ]), shallowEqual);
+
+    const style = getStyleSheet(theme);
+
+    const renderWebView = () => {
+        const userAgent = 'Mozilla/5.0 (Linux; Android 10; Android SDK built for x86 Build/LMY48X) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/81.0.4044.117 Mobile Safari/608.2.11';
+        return (
+            <WebView
```

Comment:
it doesn't look like we need to retrieve any cookies after this, can we make this incognito?

@jupenur to keep in mind, would you validate this please?

Reply:
Doesn't this need access to the cookie jar to get the Mattermost session?

Reply:
I'm not entirely sure but it seems this is going to be used to authenticate with the plugin or link the accounts or something similar. Thus I think we need to be careful and hopefully use an incognito sort of session if possible

Reply:
As jupenur said, we most certainly need the cookie jar for the Mattermost session. Also, as far as I see in the definition of WebView, the incognito prop is only for iOS devices, so apart of not working the same way in Android, I would not be able to test it.